docs/editing-components.md

```diff
@@ -0,0 +1,376 @@
+     default to checked.
+   - Certain blocks may take other arguments, as described in the
+     [basic Wagtail blocks documentation](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#basic-block-types).
+1. [Edit the component template](#the-html-template) to do something with the
+   field's data – output it, use it to trigger a CSS class, etc.
+1. [Create a schema migration.](#schema-migrations)
+
+#### Removing a field
+
+These instructions presume that you do not care
+about any data stored in the field you are deleting.
+If that is not the case, please skip to the
+[instructions for editing a field](#editing-a-field)
+and come back here when instructed.
+
+1. Locate the field you want to remove in the block's Python class, either in
+   [atoms.py](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/v1/atomic_elements/atoms.py),
+   [molecules.py](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/v1/atomic_elements/molecules.py),
+   or [organisms.py](https://github.com/cfpb/cfgov-refresh/tree/master/cfgov/v1/atomic_elements/organisms.py).
+1. Delete the field definition.
+1. [Create a schema migration.](#schema-migrations)
+
+#### Editing a field
+
+1. [Determine if the change you want to make will need a data migration.](#you-may-also-need-a-data-migration)
+   - If the answer is **no**: make your changes,
+     [create a schema migration](#schema-migrations), and be on your merry way.
+   - If the answer is **yes**: continue on.
+1. [Add the new version of the field.](#adding-a-field)
+1. [Create a schema migration](#schema-migrations) for adding the new field.
+1. [Create a data migration](../wagtail-migrations/)
+   to copy data from the old field into the new field.
+1. [Edit the component template](#the-html-template)
+   to use the new field's data instead of the old field's data.
+1. [Remove the old field.](#removing-a-field)
+1. [Create a schema migration](#schema-migrations) for removing the old field.
+
+
+### Creating migrations for StreamField blocks
+
+**tl;dr:**
+
+1. `./cfgov/manage.py makemigrations -n <description_of_changes>`
+1. [Determine if a data migration is needed](#you-may-also-need-a-data-migration)
+
+#### Schema migrations
+
+Any time you edit a Python file related to a component,
+it's a good idea to check and see whether or not you need to include
+a schema migration file in your changes, and sometimes also a data migration.
+
+To see if a schema migration is required, run the following command
+from the root of the cfgov-refresh repository,
+within your local Docker or virtualenv environment:
+
+```bash
+./cfgov/manage.py makemigrations --dry-run
+```
+
+If anything shows up, you need a schema migration.
+Run the following, editing it to give your migration a name
+that briefly describes the change(s) you're making:
+
+```bash
+./cfgov/manage.py makemigrations -n <description_of_changes>
+```
+
+Include the created file (located in `cfgov/v1/migrations/`)
```

Comment:
As with above, if your page model that has changed (i.e. it uses a block that you've modified) is not in `v1`, the migration file will not be in `v1`. It will be in whichever Django app the page model is in.

Additionally, it's possible that if you change a `v1` model, there may be migrations that are created in other apps as a result — changing the Hero molecule will, for example, also require a migration for  `regulations3k.models.pages.RegulationLandingPage`, and that migration will live in `regulations3k/migrations`.

I would generalize this path to be wherever the page models are, and I *think* that makes more sense with the explanation of why schema migrations happen when you change a block I talked above above.

Reply:
Addressed in a47039c